02-intermediate-front-end/events-and-callbacks/readme.md

```diff
@@ -33,77 +76,592 @@ button.addEventListener("click", handleClickEvent)
+  console.log("I was clicked!")
+})
+button.addEventListener("click", function(){
+  console.log("I was clicked!")
+})
+```
+
+Those two functions look identical, so how come Javascript fired the event twice?
+
+When functions don't have a name, Javascript sort-of "makes up" a name for them to use internally. As a result, that button object probably looks something like this:
+
+```js
+var button = {
+  events: {
+    click: {
+      98x872bsjy37i: function(){
+        console.log("I was clicked!")
+      },
+      2hhj3726ksha7: function(){
+        console.log("I was clicked!")
+      }
+    },
+    mouseover: {
+
+    },
+    keydown: {
+
+    }
+    // all the other event types...
+  }
+}
+```
+
+This gives us a way of "cheating" to make `handleClickEvent` fire twice:
+
+```js
+var button = document.querySelector("button")
+var handleClickEvent = function(){
+  console.log("I was clicked!")
+}
+button.addEventListener("click", handleClickEvent)
+button.addEventListener("click", function(){
+  handleClickEvent()
 })
-app.simulateA("hover")
 ```
 
-## Color Scheme Switcher Continued
+Or with two anonymous functions:
+
+```js
+var button = document.querySelector("button")
+var handleClickEvent = function(){
+  console.log("I was clicked!")
+}
+button.addEventListener("click", function(){
+  handleClickEvent()
+})
+button.addEventListener("click", function(){
+  handleClickEvent()
+})
+```
+
+## You do: Color Scheme Switcher 
+
+Clone this repo. Remember to use the SSH URL now, instead of HTTPS!
+
+[Color Scheme Switcher](https://github.com/ga-dc/color-scheme-switcher)
+
+## The event object
+
+Back in the code we were using in-class:
+
+```js
+var button = document.querySelector("button")
+var handleClickEvent = function(){
+  console.log("I was clicked!")
+}
+button.addEventListener("click", handleClickEvent);
+```
+
+Now, you're going to make a small change by adding an argument to the anonymous function, and `console.logging` it:
+
+```js
+var button = document.querySelector("button")
+var handleClickEvent = function(e){
+  console.log("I was clicked!")
+  console.log(e)
+}
+button.addEventListener("click", handleClickEvent);
+```
+
+The `e` stands for `event`.
+
+#### Turn and talk
+
+With your partner, try clicking the button and exploring what properties the MouseEvent object contains. Look for:
+
+- A way to figure out what element was clicked on
+- A way to tell the position of the mouse when it clicked
+
+### Other events
+
+Let's explore some other events. Add a text input field into the HTML:
+
+```html
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>Events and Callbacks Practice</title>
+  </head>
+  <body>
+    <button>Click me!</button>
+    <input placeholder="Type here!" />
+    <script src="script.js"></script>
+  </body>
+</html>
+```
+
+#### You do
+
+With your partner, add an event listener for the `keyup` event to the input. Explore the Event object again. Look for:
+
+- A way to tell which key was pressed
```

Comment:
It's only important so that the have a way of figuring out which key was pressed